hack/generator/pkg/astmodel/file_definition.go

```diff
@@ -147,21 +147,10 @@ func (file *FileDefinition) generateImports() *PackageImportSet {
 	// TODO: Make this configurable
 	requiredImports.ApplyName(MetaV1PackageReference, "metav1")
 
-	// Determine if there are any conflicting imports -- these are imports with the same "name"
-	// but a different package path
-	imports := requiredImports.AsSortedSlice(ByNameInGroups)
-	for _, imp := range imports {
-		for _, otherImp := range imports {
-			if !imp.Equals(otherImp) && imp.PackageName() == otherImp.PackageName() {
-				klog.Warningf(
-					"File %v: import %v (named %v) and import %v (named %v) conflict",
-					file.packageReference,
-					imp.packageReference,
-					imp.PackageName(),
-					otherImp.packageReference,
-					otherImp.PackageName())
-			}
-		}
+	// Resolve any conflicts and report any that couldn't be fixed up automatically
+	err := requiredImports.ResolveConflicts()
+	if err != nil {
+		klog.Warningf("File %s: %v", file.packageReference, err)
```

Comment:
This should really be an error, right? The resulting code won't compile.

Reply:
Converted to use `klog.Errorf()` as discussed in today's meeting.